androidHyperskillApp/src/main/java/org/hyperskill/app/android/problems_limit/view/ui/delegate/ProblemsLimitDelegate.kt

```diff
@@ -0,0 +1,86 @@
+package org.hyperskill.app.android.problems_limit.view.ui.delegate
+
+import android.content.Context
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.core.view.updateLayoutParams
+import org.hyperskill.app.android.R
+import org.hyperskill.app.android.databinding.LayoutProblemsLimitBinding
+import org.hyperskill.app.problems_limit.presentation.ProblemsLimitFeature
+import ru.nobird.android.view.base.ui.delegate.ViewStateDelegate
+import ru.nobird.android.view.base.ui.extension.setTextIfChanged
+
+class ProblemsLimitDelegate(
+    private val viewBinding: LayoutProblemsLimitBinding,
+    private val onNewMessage: (ProblemsLimitFeature.Message) -> Unit
+) {
+
+    private var viewStateDelegate: ViewStateDelegate<ProblemsLimitFeature.ViewState>? = null
+
+    fun setup(context: Context, isDividerVisible: Boolean = false) {
+        viewBinding.problemsLimitDivider.root.isVisible = isDividerVisible
+        viewBinding.problemsLimitsContent.updateLayoutParams<ViewGroup.MarginLayoutParams> {
+            topMargin = if (isDividerVisible) {
+                context.resources.getDimensionPixelOffset(R.dimen.problems_limit_diver_margin)
+            } else {
+                0
+            }
+        }
+        viewStateDelegate = ViewStateDelegate<ProblemsLimitFeature.ViewState>().apply {
+            addState<ProblemsLimitFeature.ViewState.Idle>()
+            addState<ProblemsLimitFeature.ViewState.Content.Empty>()
+
+            addState<ProblemsLimitFeature.ViewState.Loading>(
+                viewBinding.problemsLimitSkeleton
+            )
+            if (isDividerVisible) {
+                addState<ProblemsLimitFeature.ViewState.Content.Widget>(
+                    viewBinding.problemsLimitDivider.root,
+                    viewBinding.problemsLimitsContent
+                )
+            } else {
+                addState<ProblemsLimitFeature.ViewState.Content.Widget>(
+                    viewBinding.problemsLimitsContent
+                )
+            }
+
+            if (isDividerVisible) {
+                addState<ProblemsLimitFeature.ViewState.Error>(
+                    viewBinding.problemsLimitDivider.root,
+                    viewBinding.problemsLimitRetryButton
+                )
+            } else {
+                addState<ProblemsLimitFeature.ViewState.Error>(
+                    viewBinding.problemsLimitRetryButton
+                )
+            }
+        }
+
+        viewBinding.problemsLimitRetryButton.setOnClickListener {
+            onNewMessage(
+                ProblemsLimitFeature.Message.Initialize(forceUpdate = true)
+            )
+        }
+    }
```

Comment:
It's better to move divider from ProblemsLimit ui to stepQuiz ui, where it is used and not use `isDividerVisible` flag.